### ISSUE TITLE
Parallel Model Inference with Ray Serving not working due to deprecated API

### ISSUE THREAD

**Issue by ajstewart** [kserve/kserve#3595]
/kind bug

**What steps did you take and what happened:**

I"m following the example here:

https://kserve.github.io/website/latest/modelserving/v1beta1/custom/custom_model/#parallel-model-inference

To use parallel inferencing with ray however it is crashing on startup with a message of:

```
INFO:/usr/local/lib/python3.10/site-packages/ray/serve/_private/api.py:Started Serve in namespace "serve".
2024-04-11 16:37:49 Traceback (most recent call last):
2024-04-11 16:37:49   File "/app/model.py", line 281, in <module>
2024-04-11 16:37:49     ModelServer().start({MODEL_NAME: MyModel})
2024-04-11 16:37:49   File "/usr/local/lib/python3.10/site-packages/kserve/model_server.py", line 164, in start
2024-04-11 16:37:49     models[key].deploy()
2024-04-11 16:37:49   File "/usr/local/lib/python3.10/site-packages/ray/serve/deployment.py", line 262, in deploy
2024-04-11 16:37:49     raise ValueError(
2024-04-11 16:37:49 ValueError: This API has been fully deprecated. Please use serve.run() instead.
```

I am using `KServe` v0.12 and `ray` v2.9.3 installed.

From looking at `ray` it looks like this way of running was removed in v2.8 onwards. However `KServe` is dependent on 2.9 and up on v0.12.

**What did you expect to happen:**

I expected the ray serving to work.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version: v0.12
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by sivanantha321** [comment]
This is fixed in the master #3556.

**Comment by ajstewart** [comment]
> This is fixed in the master #3556.

Ah great, thanks for pointing that out. I'll try it again in the next release!

**Comment by sivanantha321** [comment]
The new patch release 0.12.1 has fix for this.
